src/bot.rs

```diff
@@ -16,51 +10,88 @@ mod del;
-    Help,
-    #[command(description = "Adiciona um alarme para um item, e notifica uma série de usuários")]
-    Add(String),
-    #[command(description = "Lista os items monitorados")]
-    List,
-    #[command(description = "Deleta um alarme")]
-    Del(String),
-    #[command(description = "Pede dinheiro emprestado pro Hulk Agiota")]
-    Lend(String),
-}
+use futures::StreamExt;
+use telegram_bot::{
+    types::{MessageKind, UpdateKind},
+    Api, CanReplySendMessage, Message, SendMessage,
+};
+
+pub async fn run() -> Result<(), Error> {
+    let bot_token = env::var("BOT_TOKEN").expect("BOT_TOKEN not present on environment");
+    let bot_username = env::var("BOT_USERNAME").expect("BOT_USERNAME not present on environment");
 
-async fn answer(cx: UpdateWithCx<AutoSend<Bot>, Message>, command: Command) -> Result<(), Error> {
     let db = db::get_db().await?;
 
-    match command {
-        Command::Help => {
-            cx.answer(Command::descriptions()).send().await?;
-        }
-        Command::Add(input) => {
-            info!("{}", cx.update.text().unwrap());
-            add::handler(cx, input, db).await?
-        }
-        Command::List => list::handler(cx, db).await?,
-        Command::Del(input) => del::handler(cx, input, db).await?,
-        Command::Lend(input) => lend::handler(cx, input).await?,
-    };
+    info!("Starting bot");
+
+    let api = Api::new(bot_token);
+
+    let mut stream = api.stream();
+    while let Some(update) = stream.next().await {
+        let update = update?;
+
+        match update.kind {
+            UpdateKind::Message(message) => handle_message(message, &api, &db, &bot_username).await,
+            UpdateKind::EditedMessage(_) => Ok(()),
+            UpdateKind::ChannelPost(_) => Ok(()),
+            UpdateKind::EditedChannelPost(_) => Ok(()),
+            UpdateKind::InlineQuery(_) => Ok(()),
+            UpdateKind::CallbackQuery(_) => Ok(()),
+            UpdateKind::Poll(_) => Ok(()),
+            UpdateKind::PollAnswer(_) => Ok(()),
+            UpdateKind::Error(_) => Ok(()),
+            UpdateKind::Unknown => Ok(()),
+        }?;
+    }
 
     Ok(())
 }
 
-pub async fn run() -> Result<(), Error> {
-    let bot_username = env::var("BOT_USERNAME").expect("BOT_USERNAME not present on environment");
+async fn handle_message(
+    message: Message,
+    api: &Api,
+    db: &Database,
+    bot_username: &str,
+) -> Result<(), Error> {
+    if let MessageKind::Text { ref data, .. } = message.kind {
+        if !data.starts_with('/') {
+            return Ok(());
+        }
 
-    info!("Starting bot");
+        info!("<{}>: {}", &message.from.first_name, data);
+
+        let first_space = data.find(' ').unwrap_or_else(|| data.len());
 
-    let bot = Bot::from_env().auto_send();
+        let (cmd, _) = data.split_at(first_space);
 
-    teloxide::commands_repl(bot, bot_username, answer).await;
+        match cmd.replace(bot_username, "").as_str() {
```

Comment:
What if there is no `@`?

Reply:
Then it would return the whole string: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d9319b16d5a22e132998ada16cb89a68)